x-pack/plugins/security_solution/public/common/components/exceptions/viewer/index.tsx

```diff
@@ -4,96 +4,411 @@
+      dispatch({
+        type: 'updateFilterOptions',
+        filterOptions: filter,
+        pagination: pag,
+      });
+    },
+    [dispatch]
+  );
+
+  const onAddException = useCallback(
+    (type: ExceptionListType): void => {
+      setIsModalOpen(true);
+    },
+    [setIsModalOpen]
+  );
+
+  const onEditExceptionItem = useCallback(
+    (exception: ExceptionListItemSchema): void => {
+      // TODO: Added this just for testing. Update
+      // modal state logic as needed once ready
+      dispatch({
+        type: 'updateExceptionToEdit',
+        exception,
+      });
 
-  const onEdit = useCallback((): void => {
-    handleEdit(exceptionItem);
-  }, [handleEdit, exceptionItem]);
+      setIsModalOpen(true);
+    },
+    [setIsModalOpen]
+  );
 
-  const onCommentsClick = useCallback((): void => {
-    setShowComments(!showComments);
-  }, [setShowComments, showComments]);
+  const onCloseExceptionModal = useCallback(
+    ({ actionType, listId }): void => {
+      setIsModalOpen(false);
 
-  const formattedComments = useMemo((): EuiCommentProps[] => {
-    return getFormattedComments(exceptionItem.comments);
-  }, [exceptionItem]);
+      // TODO: This callback along with fetchList can probably get
+      // passed to the modal for it to call itself maybe
+      if (actionType === ModalAction.CREATE && listId != null && onAssociateList != null) {
+        onAssociateList(listId);
+      }
+
+      onFetchList();
+    },
+    [setIsModalOpen, onFetchList, onAssociateList]
+  );
+
+  const setLoadingItemIds = useCallback(
+    (items: ApiProps[]): void => {
+      dispatch({
+        type: 'updateLoadingItemIds',
+        items,
+      });
+    },
+    [dispatch]
+  );
+
+  const onDeleteException = useCallback(
+    ({ id, namespaceType }: ApiProps) => {
+      deleteExceptionItem({
+        id,
+        namespaceType,
+        onSuccess: () => {
+          setLoadingItemIds(loadingItemIds.filter((t) => t.id !== id));
+          onFetchList();
+        },
+        onError: () => {
+          const dispatchToasterError = onDispatchToaster({
+            color: 'danger',
+            title: i18n.DELETE_EXCEPTION_ERROR,
+            iconType: 'alert',
+          });
+
+          dispatchToasterError();
+          setLoadingItemIds(loadingItemIds.filter((t) => t.id !== id));
+        },
+      });
+    },
+    [setLoadingItemIds, deleteExceptionItem, loadingItemIds, onFetchList, onDispatchToaster]
+  );
+
+  // Logic for initial render
+  useEffect((): void => {
+    if (initLoading && !loadingList && (exceptions.length === 0 || exceptions != null)) {
+      setInitLoading(false);
+    }
+  }, [initLoading, exceptions, loadingList]);
+
+  const ruleSettingsUrl = useMemo((): string => {
+    return services.application.getUrlForApp(
+      `security#/detections/rules/id/${encodeURI(ruleId)}/edit`
+    );
+  }, [ruleId, services.application]);
+
+  const exceptionsSubtext = useMemo((): JSX.Element => {
```

Comment:
nit: Since this component is getting a bit larger you could pull these out to their own file since they're pretty self-contained.

Also, thoughts on displaying this text right-aligned in the `UtilityBar` to remove the bounce?

<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/84169083-6c964f00-aa35-11ea-8060-9d80caabb9e7.gif" />
</p>

To something like:
<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/84169147-8172e280-aa35-11ea-9a05-048450807445.png" />
</p>

Reply:
Also noticed a bit of lag when first enabling/disabling a filter vs toggling between the two filters. No need to address in this PR, just commenting for when you loop back around once the API dependencies are resolved. Here's a perf chart of those actions:

<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/84179957-f77e4600-aa43-11ea-8a83-2c08b40a8761.png" />
</p>

Reply:
I believe that's because right now I'm not making an api call when toggling. Were I to add that, I think they would match. This is a good find though. I'll jot it down to look at once the api's are updated since I think right now my workaround is part of the slowness.

Reply:
I update the `UtilityBar` to incorporate your suggestion in a follow up PR as this one is getting big.